planner/cascades/testdata/stringer_suite_out.json

```diff
@@ -0,0 +1,177 @@
+[
+  {
+    "Name": "TestGroupStringer",
+    "Cases": [
+      {
+        "SQL": "select b from t where a > 1 and b < 1",
+        "Result": [
+          "Group#0 Schema:[Column#13]",
+          "    Projection_3 input:[Group#1], Column#2",
+          "Group#1 Schema:[Column#1,Column#2]",
+          "    TableGather_5 input:[Group#2]",
+          "Group#2 Schema:[Column#1,Column#2]",
+          "    Selection_8 input:[Group#3], lt(Column#2, 1)",
```

Comment:
The relation among groups is organized like a tree. Can we present it by a tree, like the explanation which in `planner.core`?

Reply:
Actually, groups are organized as a DAG. A Group can be referenced by more than one Group. This PR doesn't show a case of DAG because we have not implemented the tranformation rules >,<.
I think this is a common format for cascades like planner, here is the format which calcite uses:
```
Original rel:
LogicalFilter(condition=[>($0, 1941)]): rowcount = 50.0, cumulative cost = {250.0 rows, 301.0 cpu, 0.0 io}, id = 2375
  LogicalProject(k=[$2]): rowcount = 100.0, cumulative cost = {200.0 rows, 201.0 cpu, 0.0 io}, id = 2374
    LogicalTableScan(table=[[s, beatles2]]): rowcount = 100.0, cumulative cost = {100.0 rows, 101.0 cpu, 0.0 io}, id = 2358

Sets:
Set#0, type: RecordType(INTEGER i, VARCHAR j, INTEGER k)
    rel#2376:Subset#0.NONE.[], best=null, importance=0.8019000000000001
        rel#2358:LogicalTableScan.NONE.[](table=[s, beatles2]), rowcount=100.0, cumulative cost={inf}
Set#1, type: RecordType(INTEGER k)
    rel#2378:Subset#1.NONE.[], best=null, importance=0.81
        rel#2377:LogicalProject.NONE.[](input=RelSubset#2376,k=$2), rowcount=100.0, cumulative cost={inf}
    rel#2402:Subset#1.ENUMERABLE.[], best=null, importance=0.405
Set#2, type: RecordType(INTEGER k)
    rel#2380:Subset#2.NONE.[], best=null, importance=0.99
        rel#2379:LogicalFilter.NONE.[](input=RelSubset#2378,condition=>($0, 1941)), rowcount=50.0, cumulative cost={inf}
        rel#2410:LogicalProject.NONE.[](input=RelSubset#2409,k=$2), rowcount=50.0, cumulative cost={inf}
    rel#2383:Subset#2.ENUMERABLE.[], best=null, importance=1.0
        rel#2384:AbstractConverter.ENUMERABLE.[](input=RelSubset#2380,convention=ENUMERABLE,sort=[]), rowcount=50.0, cumulative cost={inf}
        rel#2403:EnumerableFilter.ENUMERABLE.[](input=RelSubset#2402,condition=>($0, 1941)), rowcount=50.0, cumulative cost={inf}
        rel#2413:EnumerableProject.ENUMERABLE.[](input=RelSubset#2412,k=$2), rowcount=50.0, cumulative cost={inf}
Set#3, type: RecordType(INTEGER i, VARCHAR j, INTEGER k)
    rel#2409:Subset#3.NONE.[], best=null, importance=0.891
        rel#2406:LogicalFilter.NONE.[](input=RelSubset#2376,condition=>($2, 1941)), rowcount=50.0, cumulative cost={inf}
    rel#2412:Subset#3.ENUMERABLE.[], best=null, importance=0.4455
    rel#2419:Subset#3.BINDABLE.[], best=rel#2417, importance=0.4455
        rel#2417:BindableTableScan.BINDABLE.[](table=[s, beatles2],filters=[>($2, 1941)]), rowcount=100.0, cumulative cost={0.5 rows, 0.505 cpu, 0.0 io}
```